plaso/helpers/macos/location.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+"""MacOS Core Location services helper."""
+import os
+
+from plaso.lib.aul import constants
+from plaso.lib import dtfabric_helper
+from plaso.lib import errors
+
+
+class ClientAuthStatusHelper(object):
+  """Core Location Client Authorisation Status helper"""
+  _AUTH_STATUS_CODES = {
+    0: 'Not Determined',
+    1: 'Restricted',
+    2: 'Denied',
+    3: 'Authorized Always',
+    4: 'Authorized When In Use'}
+
+  @classmethod
+  def GetCode(cls, code):
+    """Retrieves the description for a code.
+
+    Args:
+      code (int): status code
+
+    Returns:
+      str: name of the status code.
+    """
+    return cls._AUTH_STATUS_CODES.get(code, 'UNKNOWN: {0:d}'.format(code))
+
+
+class DaemonStatusHelper(object):
+  """Core Location Daemon Status helper"""
+  _DAEMON_STATUS_CODES = {
+    0: 'Reachability Unavailable',
+    1: 'Reachability Small',
+    2: 'Reachability Large',
+    56: 'Reachability Unachievable'}
+
+  @classmethod
+  def GetCode(cls, code):
+    """Retrieves the description for a code.
+
+    Args:
+      code (int): status code
+
+    Returns:
+      str: name of the status code.
+    """
+    return cls._DAEMON_STATUS_CODES.get(code, 'UNKNOWN: {0:d}'.format(code))
+
+
+class SubharvesterIDHelper(object):
+  """Core Location Subharvester ID helper"""
+  _SUBHARVESTER_ID = {
+    1: 'Wifi',
+    2: 'Tracks',
+    3: 'Realtime',
+    4: 'App',
+    5: 'Pass',
+    6: 'Indoor',
+    7: 'Pressure',
+    8: 'Poi',
+    9: 'Trace',
+    10: 'Avenger',
+    11: 'Altimeter',
+    12: 'Ionosphere',
+    13: 'Unknown'}
+
+  @classmethod
+  def GetCode(cls, subharvester_id):
+    """Retrieves the description for an ID.
+
+    Args:
+      subharvester_id (int): identifier
+
+    Returns:
+      str: name of the ID.
+    """
+    return cls._SUBHARVESTER_ID.get(
+        subharvester_id, 'UNKNOWN: {0:d}'.format(subharvester_id))
+
+
+class LocationClientStateTrackerParser(dtfabric_helper.DtFabricHelper):
+  """LocationClientStateTracker data chunk parser"""
+
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), 'location.yaml')
+
+  def Parse(self, data):
+    """Parses given data of a given size as a LocationClientStateTracker chunk
+
+    Args:
+      data (bytes): Raw data
+
+    Returns:
+      Dict: The state tracker data
+
+    Raises:
+      ParseError: if the data cannot be parsed.
+    """
+    return self._ReadStructureFromByteStream(
+      data, 0, self._GetDataTypeMap('location_tracker_client_data')).__dict__
```

Comment:
Why not just return the `state tracker data` value object instead of its internal dict?

Reply:
I believe I did this because it made it easier/neater to merge the dicts during later formatting